apps/jupyter/templates/services/app/route.yml.j2

```diff
@@ -0,0 +1,27 @@
+apiVersion: v1
+kind: Route
+metadata:
+  namespace: "{{ project_name }}"
+  name: jupyter
+  labels:
+    app: jupyter
+    service: app
+    version: "{{ jupyter_image_tag }}"
+  annotations:
+    haproxy.router.openshift.io/disable_cookies: "true"
+    haproxy.router.openshift.io/balance: "roundrobin"
+{% if jupyter_app_ip_whitelist | length %}
+{% for ip in jupyter_app_ip_whitelist %}
+    haproxy.router.openshift.io/ip_whitelist: "{{ ip }}"
```

Comment:
Are you sure this is working ? Can you multiple time the same annotations ? Reading the documentation I'm not sure : https://docs.openshift.com/dedicated/3/architecture/networking/routes.html#whitelist
Does it work behind a proxy ?

Reply:
My implementation was broken, thanks! And Yes: it works behind a proxy.